tools/sdk/tribeca/idls/quarry_mine.ts

```diff
@@ -0,0 +1,1963 @@
+          index: false
+        }
+      ]
+    },
+    {
+      name: 'RewarderAnnualRewardsUpdateEvent'
+      fields: [
+        {
+          name: 'previousRate'
+          type: 'u64'
+          index: false
+        },
+        {
+          name: 'newRate'
+          type: 'u64'
+          index: false
+        },
+        {
+          name: 'timestamp'
+          type: 'i64'
+          index: false
+        }
+      ]
+    },
+    {
+      name: 'MinerCreateEvent'
+      fields: [
+        {
+          name: 'authority'
+          type: 'publicKey'
+          index: false
+        },
+        {
+          name: 'quarry'
+          type: 'publicKey'
+          index: false
+        },
+        {
+          name: 'miner'
+          type: 'publicKey'
+          index: false
+        }
+      ]
+    },
+    {
+      name: 'QuarryCreateEvent'
+      fields: [
+        {
+          name: 'tokenMint'
+          type: 'publicKey'
+          index: false
+        },
+        {
+          name: 'timestamp'
+          type: 'i64'
+          index: false
+        }
+      ]
+    },
+    {
+      name: 'QuarryRewardsUpdateEvent'
+      fields: [
+        {
+          name: 'tokenMint'
+          type: 'publicKey'
+          index: false
+        },
+        {
+          name: 'annualRewardsRate'
+          type: 'u64'
+          index: false
+        },
+        {
+          name: 'rewardsShare'
+          type: 'u64'
+          index: false
+        },
+        {
+          name: 'timestamp'
+          type: 'i64'
+          index: false
+        }
+      ]
+    }
+  ]
+  errors: [
+    {
+      code: 6000
+      name: 'Unauthorized'
+      msg: 'You are not authorized to perform this action.'
+    },
+    {
+      code: 6001
+      name: 'InsufficientBalance'
+      msg: 'Insufficient staked balance for withdraw request.'
+    },
+    {
+      code: 6002
+      name: 'PendingAuthorityNotSet'
+      msg: 'Pending authority not set'
+    },
+    {
+      code: 6003
+      name: 'InvalidRewardsShare'
+      msg: 'Invalid quarry rewards share'
+    },
+    {
+      code: 6004
+      name: 'InsufficientAllowance'
+      msg: 'Insufficient allowance.'
+    },
+    {
+      code: 6005
+      name: 'NewVaultNotEmpty'
+      msg: 'New vault not empty.'
+    },
+    {
+      code: 6006
+      name: 'NotEnoughTokens'
+      msg: 'Not enough tokens.'
+    },
+    {
+      code: 6007
+      name: 'InvalidTimestamp'
+      msg: 'Invalid timestamp.'
+    },
+    {
+      code: 6008
+      name: 'InvalidMaxClaimFee'
+      msg: 'Invalid max claim fee.'
+    },
+    {
+      code: 6009
+      name: 'MaxAnnualRewardsRateExceeded'
+      msg: 'Max annual rewards rate exceeded.'
+    },
+    {
+      code: 6010
+      name: 'Paused'
+      msg: 'Rewarder is paused.'
+    },
+    {
+      code: 6011
+      name: 'UpperboundExceeded'
+      msg: "Rewards earned exceeded quarry's upper bound."
+    }
+  ]
+}
+export const QuarryMineJSON: QuarryMineIDL = {
```

Comment:
This content is not JSON, would be better to put that in a JSON file, import it here and use it with `generateMapErrors` IMO

Reply:
tbh I copied theses IDL files directly from tribeca source code iirc. Would tend to keep it that way, but we could refactor it. I have never used `generateMapErrors` and would appreciate your help to do so :)